Transform RecoTrackRefSelector to stream::EDProducer

makortel opened the pull request:
This PR changes RecoTrackRefSelector to a stream::EDProducer (using ObjectSelectorStreamProducer introduced in #9904). (for history, the PR first contained only a simple transformation to stream::EDFilter, but was then extended to contain the change to producer as there we re no use cases for event filtering)

Tested in 7_6_0_pre3, no changes expected in results.

@rovere @VinInn @fwyzard

cmsbuild commented:
A new Pull Request was created by @makortel (Matti Kortelainen) for CMSSW_7_6_X.

Transform RecoTrackRefSelector to stream

It involves the following packages:

CommonTools/RecoAlgos


@cmsbuild, @cvuosalo, @slava77 can you please review it and eventually sign? Thanks.
@ahinzmann, @jdolen, @rappoccio, @abbiendi, @jhgoh this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

makortel commented:
@fwyzard I guess you want this backported, both 74X and 75X?

fwyzard commented:
Hi Matti,
just 7.5.x for the time being.

Thank you,
.Andrea

makortel commented:
@fwyzard OK, 75X backport is in #10912.

fwyzard commented:
Hi Matti,
changing this from an EDFilter to an EDProducer is OK for the HLT, as we only use it as a producer anyway.
Fixing the HLT configuration via `customizeHLTforCMSSW.py` should be OK.

.A

makortel commented:
Hi Andrea,
thanks for the confirmation. Would it be enough to change it as EDProducer in 76X (the supporting #9904 is in 76X only), or is it necessary to backport that change too to 75X? (I'd prefer the former, but latter is also doable)

Cheers,
Matti

fwyzard commented:
For HLT I would prefer to make the change to a producer only in 76x, too.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

makortel committed:
Transform RecoTrackRefSelector to EDProducer

cmsbuild commented:
Pull request #10911 was updated. @perrotta, @cmsbuild, @cvuosalo, @slava77, @Martin-Grunewald, @fwyzard can you please check and sign again.

makortel commented:
I included also the EDProducer transformation (so now its stream::EDProducer). If this is deemed ok, I'll update the PR title and description, and possibly squash the commits together.

cmsbuild commented:
-1 
Tested at: 2416da4f25f7b66d952c5b053ea82e43e3b0121c
 When I ran the RelVals I found an error in the following worklfows: 
 4.22 step1
<pre>DAS Error</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7575/summary.html

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 38d790639aa89b4d8c3c7591d9c627ce89950d4d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7599/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7599/summary.html

cvuosalo commented:
+1

For #10911 38d790639aa89b4d8c3c7591d9c627ce89950d4d

Making RecoTrackRefSelector a stream module. There should be no change in monitored quantities.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7599/summary.html) against baseline CMSSW_7_6_X_2015-08-24-1100 show no significant differences, as expected.

makortel commented:
@fwyzard Sure, I can do that. But I'm unable to do it before evening (or it may slip to Monday).

makortel commented:
@fwyzard There are two other `customiseFor####` functions in `customizeHLTforCMSSW.py` that modify RecoTrackRefSelector parameters by the module label (hltBSoftMuonMu5L3). Should I change also those to the generic style in this PR? (I don't see much point in fixing only one of them)

fwyzard commented:
Ah, I see... let's keep it like this so far, we will change the customisation if we need to add more copies of the module.

fwyzard commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs  (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

slava77 commented:
@makortel this doesn't merge anymore
Please check

makortel commented:
Rebased on top of CMSSW_7_6_X_2015-08-30-1100 (`customizeHLTforCMSSW.py` was changed in HEAD). Also modified the title and description to better reflect what the PR contains.

cmsbuild commented:
Pull request #10911 was updated. @perrotta, @cmsbuild, @cvuosalo, @slava77, @Martin-Grunewald, @fwyzard can you please check and sign again.

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0ebd0b6d86e81e35731514d8a1aa4247f9ebffe7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7773/summary.html

Martin-Grunewald commented:
+1

slava77 commented:
There is a merge conflict here
Please fix

makortel commented:
The merge conflict follows the merge+automatic forward port of #10912. Git merge apparently does not "realize" that the same commit is also here. I'll rebase after we get an 76X IB containing #10912.

makortel commented:
Rebased on top of CMSSW_7_6_X_2015-09-02-1100 (note that the first commit disappeared as that got included in #10912).

fwyzard commented:
+1

cmsbuild commented:
Pull request #10911 was updated. @cmsbuild, @cvuosalo, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 53569f8480096afe883244af6528787c742923f6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7834/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10911/7834/summary.html

cvuosalo commented:
+1

For #10911 53569f8480096afe883244af6528787c742923f6

This PR was already approved by Reco, but a merge conflict developed. It was rebased. Jenkins tests are still OK, except for workflow 50202.0 that encountered some error. A short test of 50202.0 with 15 events against baseline CMSSW_7_6_X_2015-09-02-1100 shows no significant differences, so the Jenkins error is not related to this PR.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs  (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1
